pkgs/applications/networking/irc/weechat/default.nix

```diff
@@ -26,6 +34,16 @@ let
   ];
   enabledPlugins = builtins.filter (p: p.enabled) plugins;
 
+  aspellDictEnv = buildEnv{
```

Comment:
Can you use `aspellWithDicts`?

Reply:
Good idea, I don't think the same can be done for the enchant/hunspell pair without a bit more plumbing though.

Reply:
@FRidh done

Reply:
> Why not just install them globally with an enchant module, instead of overriding each package separately?

One may want to be explicit about what dictionaries are used.

But I agree with you; in practice one likely just wants what is provided by the system (`XDG_DATA_DIRS`).

Reply:
That didn't seem to be working in my case, which motivated this PR. I think there's value in having this all self contained though, indeed I'm currently running weechat on a server with a user without XDG_DATA_DIRS.

I do agree that patching each user of `enchant` isn't a perfect solution though. Installing them globally doesn't quite fit the nix philosophy though. A much better change would be to patch aspell so that not only the binary can find the dictionaries in `aspellWithDicts` but the library can too (and similarly for hunspell).

Reply:
One small downside of that is that spelling dictionaries seem to be a fairly fragile part of nixpkgs, and meddling with aspell/hunspell internals might have to be done carefully.